### ISSUE TITLE
Error in OpenCL calculation: Update your GPU drivers or reduce the amount of games played simultaneously. 

### ISSUE THREAD

**Issue by jiangwuyou** [pnprog/goreviewpartner#92]
Compiled the latest code and release it at Chinese Go BBS.

One Go player met that issue: Intel i7-8550U 1.99,  Intel UHD Graphics 620, 8GB memory.
It has been set 4 configurations with different weight files.  It allowes only the first configuration for Leela Zero to work normally, other configuration will fail.  That PC supports Half precision compute.

I tested the same 4 configrations on two older computers. No issue.

Using 2 thread(s). 
RNG seed: 9316522856599505871 
BLAS Core: Haswell 
Detecting residual layers...v2...224 channels...20 blocks. 
Initializing OpenCL (autodetecting precision). 
Detected 1 OpenCL platforms. 
Platform version: OpenCL 2.1  
Platform profile: FULL_PROFILE 
Platform name:    Intel(R) OpenCL 
Platform vendor:  Intel(R) Corporation 
Device ID:     0 Device name:   Intel(R) UHD Graphics 620 
Device type:   GPU Device vendor: Intel(R) Corporation 
Device driver: 25.20.100.6326 
Device speed:  1150 MHz 
Device cores:  24 CU 
Device score:  621 
Device ID:     1 
Device name:   Intel(R) Core(TM) i7-8550U CPU @ 1.80GHz 
Device type:   CPU 
Device vendor: Intel(R) Corporation 
Device driver: 7.6.0.716 
Device speed:  1800 MHz 
Device cores:  8 CU 
Device score:  521 
Selected platform: Intel(R) 
OpenCL Selected device: Intel(R) UHD Graphics 620 with OpenCL 2.1 capability. 
Half precision compute support: Yes.
 OpenCL: using fp16/half compute support. 
Loaded existing SGEMM tuning. 
Wavefront/Warp 
size: 16 Max workgroup 
size: 256 Max workgroup 
dimensions: 256 256 256  
OpenCL: fp16/half failed despite driver claiming support. 
Falling back to single precision 
Detected 1 OpenCL platforms. 
Platform version: OpenCL 2.1  
Platform profile: FULL_PROFILE 
Platform name:    Intel(R) OpenCL 
Platform vendor:  Intel(R) 
Corporation Device ID:     0 
Device name:   Intel(R) UHD Graphics 620 
Device type:   GPU Device vendor: Intel(R) Corporation 
Device driver: 25.20.100.6326 
Device speed:  1150 MHz 
Device cores:  24 CU 
Device score:  621 
Device ID:     1 
Device name:   Intel(R) Core(TM) i7-8550U CPU @ 1.80GHz Device type:   CPU 
Device vendor: Intel(R) Corporation 
Device driver: 7.6.0.716 
Device speed:  1800 MHz 
Device cores:  8 CU 
Device score:  521 
Selected platform: Intel(R) OpenCL 
Selected device: Intel(R) UHD Graphics 620 with OpenCL 2.1 capability. 
Half precision compute support: Yes.
Loaded existing SGEMM tuning.
Wavefront/Warp size: 16 Max workgroup 
size: 256 Max workgroup 
dimensions: 256 256 256  
Setting max tree size to 4136 MiB and cache size to 459 MiB

**Comment by pnprog** [comment]
Hi @jiangwuyou ,

Several users of LeelaZero seems to have this issue, not a LeelaZero issue, not GRP issue. Most probably:

> Hardware, overclocking or driver issue

Ref: https://github.com/gcp/leela-zero/issues/776

**Comment by pnprog** [comment]
Maybe your friend should go for the CPU version of LeelaZero.

**Comment by jiangwuyou** [comment]
Yeah. After reinstalled the driver, it can work always the first configuration, but failed always with other configuraiton files. GoReviewPartner refuse to start the Go play with "Error in OpenCL calculation" message. It's possible to let user to play with the buggy Intel 620 display card?

**Comment by pnprog** [comment]
Maybe you can try a different version of LeelaZero. The latest version 0.16 made some change to use different precision (full or half) with purpose to works better with old GPU but maybe it breaks compatibility. So try with both v0.15 and v0.16 to see if there is a difference.

The 3 other configurations (that are not working) only use different version of the weights file? but same LeelaZero executable? maybe you could share the content of **config.ini** to check in details.

By the way, what is **Chinese Go BBS** ?

**Comment by jiangwuyou** [comment]
>The latest version 0.16 

The issue happened with v0.16 .

>what is Chinese Go BBS ?

It's at https://tieba.baidu.com/f?kw=%E5%9B%B4%E6%A3%8B&ie=utf-8

>The 3 other configurations (that are not working) only use different version of the weights file? but same LeelaZero executable? maybe you could share the content of config.ini to check in details.

It's here:

[General]
language = zh
sgffolder = D:/
livefolder = 
rsgffolder = D:/

[Leela]
analysisbot = slow
slowcommand = 
liveanalysisbot = both
fastcommand = 
slowparameters = --gtp --noponder
slowtimepermove = 15
fastparameters = --gtp --noponder
fasttimepermove = 5
liveplayerbot = fast
reviewbot = fast

[AQ]
analysisbot = slow
slowcommand = 
liveanalysisbot = both
fastcommand = 
liveplayerbot = fast
slowparameters = 
fastparameters = 
reviewbot = fast

[Ray]
analysisbot = slow
slowcommand = 
liveanalysisbot = both
fastcommand = 
liveplayerbot = fast
slowparameters = --no-gpu --const-time 15
fastparameters = --no-gpu --const-time 5
reviewbot = fast

[GnuGo]
analysisbot = slow
slowcommand = 
liveanalysisbot = both
fastcommand = 
liveplayerbot = fast
variations = 4
deepness = 4
slowparameters = --mode=gtp --level=15
fastparameters = --mode=gtp --level=10
reviewbot = fast

[LeelaZero]
analysisbot = both
slowcommand = ..\leela\leelaz.exe
liveanalysisbot = both
fastcommand = ..\leela\leelaz.exe
slowparameters = --gtp --noponder --weights elf.txt
slowtimepermove = 20
fastparameters = --gtp --noponder --weights weights.txt
fasttimepermove = 10
liveplayerbot = both
reviewbot = both

[Analysis]
maxvariations = 26
novariationifsamemove = False
savecommandline = False
stopatfirstresign = False
analyser = LeelaZero - 腐败狗最新版

[Review]
fuzzystoneplacement = 0.2
realgamesequencedeepness = 5
gobanscreenratio = 0.5
maxvariations = 26
variationscoloring = blue_for_winning
variationslabel = rate
invertedmousewheel = False
lastgraph = 点目
leftgobanratio = 0.395833333333
rightgobanratio = 0.395833333333
lastmap = 
rightpanelratio = 0.399264705882
yellowbar = #F39C12
lastbot = 

[Live]
nooverlap = False
size = 19
komi = 7.5
handicap = 0
analyser = LeelaZero - 腐败狗最新版
black = 人
white = LeelaZero - 腐败狗最新版
livegobanratio = 0.791666666667

[LeelaZero-0]
profile = 里拉狗最新版
command = ../Leela/leelaz.exe
parameters = --gtp --noponder --weights weights.txt
timepermove = 15

[LeelaZero-1]
profile = 腐败狗最新版
command = ../Leela/leelaz.exe
parameters = --gtp --noponder --weights elf.txt
timepermove = 15

[LeelaZero-2]
profile = 里拉狗低配版
command = ../Leela/leelaz.exe
parameters = --gtp --noponder --weights leela157.txt
timepermove = 15

[LeelaZero-3]
profile = 腐败狗旧版
command = ../Leela/leelaz.exe
parameters = --gtp --noponder --weights elf20180504.txt
timepermove = 15
